data/common.yaml

```diff
@@ -0,0 +1,15 @@
+---
+
```

Comment:
the data here is duplicated and the file can be deleted? the content is (and also should) be set in the init.pp

Reply:
I disagree. The data is only at the init.pp, because ```puppetlabs/puppetlabs-strings``` sucks and can not handle Hiera (data in modules) properly.

Reply:
but having them at two places is a bad approach. It's not clear which is the source of truth. People will always assume that the first file is correct and never update the second one as well. We already did that in multiple modules. For example: https://github.com/voxpupuli/puppet-ferm/commit/53cfc040f415d819a0feaaf7b5c8e9af6ac6c079

Reply:
I agree with @bastelfreak, the default values should be in the code.

Reply:
I am pretty sure there are other repositories, where we maintain the values at both places. This was supposed to be a workaround, until ```puppetlabs/puppetlabs-strings``` will grab the values from ```data/common.yaml```.

Reply:
Just a quick overview: 
```
./puppet-autofs/data/common.yaml
./puppet-borg/data/common.yaml
./puppet-cron/data/common.yaml
./puppet-fail2ban/data/common.yaml
./puppet-grafana/data/common.yaml
./puppet-ipset/data/common.yaml
./puppet-jail/data/common.yaml
./puppet-letsencrypt/data/common.yaml
./puppet-lldpd/data/common.yaml
./puppet-mlocate/data/common.yaml
./puppet-nscd/data/common.yaml
./puppet-nsd/data/common.yaml
./puppet-prosody/data/common.yaml
./puppet-puppetwebhook/data/common.yaml
./puppet-rabbitmq/data/common.yaml
./puppet-rhsm/data/common.yaml
./puppet-rsyslog/data/common.yaml
./puppet-selinux/data/common.yaml
./puppet-unbound/data/common.yaml
./puppet-yum/data/common.yaml
```

Some of them do not even have the defaults at ```init.pp``` which will be a problem regarding ```puppetlabs/puppetlabs-strings```.

Reply:
This is just my .02, and it is worth even less these days, but I actually had to run a test to determine which value would take precedence. I worked with groups that were pro code duplication, and those that were not. My experience and mindset believe code duplication is evil, as I have too many time in my long career fixed bug multiple time due to duplication.

Reply:
I'm strongly in favour of not putting values into hiera unless they're overriding defaults.

Reply:
We should write this somewhere down, to prevent almost always having this discussion. :(

Reply:
And the final answer is?

Reply:
Just drop the duplicate stuff from the ```data/common.yaml```, as you did in the first place. :(